+nosnoc/+discrete_problem/stewart.m

```diff
@@ -0,0 +1,771 @@
+                        lambda_lambda = obj.w.lambda_lambda(ii,jj);
+                        lambda_theta = obj.w.lambda_theta(ii,jj);
+                        B_max = obj.w.B_max(ii,jj);
+                        pi_lambda = obj.w.pi_lambda(ii,jj);
+                        pi_theta = obj.w.pi_theta(ii,jj);
+                        eta = obj.w.eta(ii,jj);
+                        nu = obj.w.nu(ii,jj);
+
+                        obj.g.pi_lambda_or(ii,jj) = {[pi_lambda-sigma_lambda_f;pi_lambda-sigma_lambda_b;sigma_lambda_f+sigma_lambda_b-pi_lambda],0,inf};
+                        obj.g.pi_theta_or(ii,jj) = {[pi_theta-sigma_theta_f;pi_theta-sigma_theta_b;sigma_theta_f+sigma_theta_b-pi_theta],0,inf};
+
+                        % kkt conditions for min B, B>=sigmaB, B>=sigmaF
+                        kkt_max = [1-lambda_theta-lambda_lambda;
+                            B_max-pi_lambda;
+                            B_max-pi_theta;
+                            (B_max-pi_lambda).*lambda_lambda - sigma;
+                            (B_max-pi_theta).*lambda_theta - sigma];
+                        obj.g.kkt_max(ii,jj) = {kkt_max,
+                            [0*ones(n_lambda,1);0*ones(n_lambda,1);0*ones(n_lambda,1);-inf*ones(n_lambda,1);-inf*ones(n_lambda,1)],
+                            [0*ones(n_lambda,1);inf*ones(n_lambda,1);inf*ones(n_lambda,1);0*ones(n_lambda,1);0*ones(n_lambda,1)]};
+
+                        % eta calculation
+                        eta_const = [eta-pi_theta;eta-pi_lambda;eta-pi_theta-pi_lambda+B_max];
+                        obj.g.eta_const(ii,jj) = {eta_const,
+                            [-inf*ones(n_lambda,1);-inf*ones(n_lambda,1);zeros(n_lambda,1)],
+                            [zeros(n_lambda,1);zeros(n_lambda,1);inf*ones(n_lambda,1)]};
+
+                        obj.g.nu_or(ii,jj) = {[nu-eta;sum(eta)-nu],0,inf};
+
+                        % the actual step eq conditions
+                        %M = 1e5;
+                        M=obj.p.T()/opts.N_stages;
+                        delta_h = obj.w.h(ii,jj) - obj.w.h(ii,jj-1);
+                        step_equilibration = [delta_h + (1/h0)*nu*M;
+                            delta_h - (1/h0)*nu*M];
+                        obj.g.step_equilibration(ii,jj) = {step_equilibration,[0;-inf],[inf;0]};
+                    end
+                end
+            end
+        end
+
+        function populate_problem(obj)
+            obj.create_variables();
+            obj.forward_sim_constraints();
+            obj.generate_complementarities();
+            obj.step_equilibration();
+
+            obj.populated = true;
+        end
+
+        function create_solver(obj, solver_options, plugin)
+            if ~obj.populated
+                obj.populate_problem()
+            end
+
+            if ~exist('plugin')
+                plugin = 'scholtes_ineq';
+            end
+            obj.w.sort_by_index();
```

Comment:
this is to get the block banded sparsity right? if yes so, please add a comment here.